misc-scripts/xref_mapping/XrefParser/BaseParser.pm

```diff
@@ -580,20 +581,9 @@ sub upload_xref_object_graphs {
 			   $xref->{SPECIES_ID},
                            'DEPENDENT');
 
-	 #####################################
-	 # find the xref_id for dependent xref. FIXME: wrong!!!
-	 # doesn't support returning multiple rows, which might be the
-	 # case e.g. for Unitprot_gn xrefs from UniProtParser
-	 #####################################
-	 $xref_id_sth->execute(
-                   $dep{ACCESSION},
-                   $dep{SOURCE_ID},
-                   $xref->{SPECIES_ID} );
-         my $dep_xref_id = ($xref_id_sth->fetchrow_array())[0];
-
-	 if(!(defined $dep_xref_id) || $dep_xref_id ==0 ){
-	   print STDERR "acc = $dep{ACCESSION} \nlink = $dep{LINKAGE_SOURCE_ID} \n".$dbi->err."\n";
-	   print STDERR "source = $dep{SOURCE_ID}\n";
+         my $dep_xref_id = $dbi->last_insert_id();
+	 if( ! $dep_xref_id ) {
```

Comment:
Smells like a tab!

Reply:
It is, that whole block has been indented with a mixture of tabs and spaces - with my additions being the only lines there only indented with the latter.

Reply:
I will forgive you eventually.